src/components/molecules/EmojiPicker/EmojiPicker.tsx

```diff
@@ -0,0 +1,64 @@
+'use client';
+
+import { useEffect, useRef } from 'react';
+import data from '@emoji-mart/data';
+import { Picker, type EmojiData } from 'emoji-mart';
+
+export interface EmojiPickerProps {
+  onEmojiSelect: (emoji: { native: string }) => void;
+  maxLength?: number;
+  currentInput?: string;
+}
+
+export function EmojiPicker({ onEmojiSelect, maxLength, currentInput }: EmojiPickerProps) {
+  const pickerRef = useRef<HTMLDivElement>(null);
+  const onEmojiSelectRef = useRef(onEmojiSelect);
+  const maxLengthRef = useRef(maxLength);
+  const currentInputRef = useRef(currentInput);
+
+  // Keep refs in sync with props
+  useEffect(() => {
+    onEmojiSelectRef.current = onEmojiSelect;
+    maxLengthRef.current = maxLength;
+    currentInputRef.current = currentInput;
+  }, [onEmojiSelect, maxLength, currentInput]);
+
+  useEffect(() => {
+    const pickerElement = pickerRef.current;
+    if (!pickerElement) return;
+
+    const handleEmojiSelect = (emojiObject: EmojiData) => {
+      const maxLen = maxLengthRef.current;
+      const current = currentInputRef.current;
+
+      if (maxLen && current) {
+        const emojiLength = new Blob([emojiObject.native]).size / 2;
```

Comment:
This measures bytes not characters. Multi-byte emojis like 👨‍💻 will break.

```suggestion
        const emojiLength = [...emojiObject.native].length;
```